### PR TITLE
Added color and typography theming.

### PR DESCRIPTION
**Pull request by KyleSaunders1310**

### CODE REVIEW

File: src/app/theme/app.theme.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+@use '~@angular/material' as material;
+@import './color-palettes.mixin.scss';
+@import './typography-config.mixin.scss';
+
+$blue-palette: $primary-palette;
```
Comment (Solenoden) [inline src/app/theme/app.theme.scss:5]: Whats going on here?

File: package.json [hunk 2]
```diff
@@ -11,10 +11,12 @@
   "private": true,
   "dependencies": {
     "@angular/animations": "~12.2.0",
+    "@angular/cdk": "^12.2.12",
```
Comment (Solenoden) [inline package.json:14]: Are you sure you want to use the ^ update type? It doesn't match the update types for the rest of your dependencies

File: src/app/theme/app.theme.scss (hunk 1 above)
Comment (KyleSaunders1310) [reply src/app/theme/app.theme.scss]: Fixed it

File: package.json (hunk 2 above)
Comment (KyleSaunders1310) [reply package.json]: Fixed it

**Review by Solenoden — APPROVED** [review]

**Pull request merged**